src/App/VPSScan/NinjaGraph.hs

```diff
@@ -107,15 +108,48 @@ generateNinjaDeps baseDir NinjaGraphOpts{..} = do
       Nothing -> "cd " ++ show baseDir ++ " && NINJA_ARGS=\"-t deps\" make"
       Just lunch ->  "cd " ++ show baseDir ++ " && source ./build/envsetup.sh && lunch " ++ T.unpack lunch ++ " && NINJA_ARGS=\"-t deps\" make"
 
-addInputsToNinjaDeps :: [DepsTarget] -> [DepsTarget]
-addInputsToNinjaDeps = map addInputToTarget
+correctedTarget :: DepsTarget -> DepsTarget
+correctedTarget target@DepsTarget { targetDependencies = [] } =
```

Comment:
I haven't seen this syntax before, what does this do? `target@DepsTarget`

Reply:
It does the same thing as just having everything after the `@` would, but also assigns the argument to a variable named `target`.